src/storage/object/delete.js

```diff
@@ -9,35 +9,54 @@
  * OF ANY KIND, either express or implied. See the License for the specific language
  * governing permissions and limitations under the License.
  */
+import { deleteFromCollab } from '../utils/collab.js';
+import { postObjectVersionWithLabel } from '../version/put.js';
 
+/**
+ * Deletes an object in the storage, creating a version of it if necessary.
+ *
+ * @param {Object} env the CloudFlare environment
+ * @param {DaCtx} daCtx the DA Context
+ * @param {String} key the key of the object to delete (excluding Org)
+ * @param {Boolean} isMove if this was initiated by a move operation
+ * @return {Promise<void>}
+ */
+export async function deleteObject(env, daCtx, key, isMove = false) {
+  const fname = key.split('/').pop();
+
+  const tmpCtx = { ...daCtx, key }; // For next calls, ctx needs the passed key, as it could contain a folder
+  if (fname.indexOf('.') > 0 && !key.endsWith('.props')) {
+    await postObjectVersionWithLabel(env, tmpCtx, isMove ? 'Moved' : 'Deleted');
+  }
+  await env.DA_CONTENT.delete(`${daCtx.org}/${key}`);
+  await deleteFromCollab(env, tmpCtx);
+}
 /**
  * Deletes one or more objects in the storage. Object is specified by the key in the daCtx or a list passed in.
  * Note: folders can not be specified in the `keys` list.
  *
  * @param {Object} env the CloudFlare environment
  * @param {DaCtx} daCtx the DA Context
- * @param {String[]} [keys=[]] the list of keys to delete (excluding the Org)
  * @return {Promise<{body: null, status: number}>}
  */
-export default async function deleteObjects(env, daCtx, keys = []) {
-  if (keys.length) {
-    const fullKeys = keys.map((key) => `${daCtx.org}/${key}`);
-    await env.DA_CONTENT.delete(fullKeys);
-    return { body: null, status: 204 };
-  }
-
-  const fullKey = `${daCtx.org}/${daCtx.key}`;
-  const prefix = `${fullKey}/`;
+export default async function deleteObjects(env, daCtx) {
+  const keys = [];
+  const prefix = `${daCtx.org}/${daCtx.key}/`;
   // The input prefix has a forward slash to prevent (drafts + drafts-new, etc.).
   // Which means the list will only pickup children. This adds to the initial list.
-  keys.push(fullKey, `${fullKey}.props`);
+  keys.push(daCtx.key, `${daCtx.key}.props`);
   let truncated = false;
   do {
-    const r2objects = await env.DA_CONTENT.list({ prefix, limit: 500 });
+    const r2objects = await env.DA_CONTENT.list({ prefix, limit: 100 });
     const { objects } = r2objects;
     truncated = r2objects.truncated;
-    keys.push(...objects.map(({ key }) => key));
-    await env.DA_CONTENT.delete(keys);
+    keys.push(...objects.map(({ key }) => key.split('/').slice(1).join('/')));
```

Comment:
Am I interpreting this correctly that it eats off the first segment of the object? I.e. the org name?

Reply:
Yep